src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -108,100 +108,143 @@ func (d *downsamplerAndWriter) Write(
 	unit xtime.Unit,
 	overrides WriteOptions,
 ) error {
-	err := d.maybeWriteDownsampler(tags, datapoints, unit, overrides)
-	if err != nil {
-		return err
+	multiErr := xerrors.NewMultiError()
+	if d.shouldDownsample(overrides) {
+		err := d.writeToDownsampler(tags, datapoints, unit, overrides)
+		if err != nil {
+			multiErr = multiErr.Add(err)
+		}
```

Comment:
I think so. I think basically its like "write to downsampler for the aggregated namespaces and then write directly to storage for unaggregated"

Although if downsampling fails you may want to avoid writing to the storage as well to prevent write amplification and out-of-order writes when retries happen, but not sure what Rob's reasoning was for changing the behavior here

Reply:
@robskillington Was there a specific reason to switch to multiErr here? I get the idea that maybe try and save some of the data if you can, but I think it could end up exacerbating performance issues when there are failures

Reply:
Mainly just that aggregation failures should be independent from writing to DB failures and also the inverse.